src/parameterTypes.ts

```diff
@@ -0,0 +1,108 @@
+import { defineParameterType } from "@cucumber/cucumber";
+import memory from "@qavajs/memory";
+
+defineParameterType({
+  name: "memory",
+  regexp: /'(.+)'/,
+  transformer: (p: string) => memory.getValue(p),
+});
+
+/**
+ * Used for returning key code by key name
+ */
+defineParameterType({
+  regexp: /"([^"\\]*(\\.[^"\\]*)*)"/,
+  transformer(keyName: string) {
+    const KEYS: any = {
+      Enter: "Enter",
+      "Ctrl+A": process.platform === "darwin" ? ["Meta", "a"] : ["\uE009", "a"],
+      Tab: "Tab",
+      ArrowRight: "ArrowRight",
+      ArrowLeft: "ArrowLeft",
+    };
+    return KEYS[keyName];
+  },
+  name: "key",
+  useForSnippets: false,
+});
+
+/**
+ * Used for returning chai assertion condition
+ */
+defineParameterType({
+  regexp:
+    /exist|be enabled|be displayed|be clickable|be focused|be displayed in viewport/,
+  transformer(condition: string) {
+    const CONDITIONS: any = {
+      exist: "waitForExist",
+      "be enabled": "waitForEnabled",
+      "be displayed": "waitForDisplayed",
+      "be clickable": "waitForClickable",
+      "be focused": "waitForElementFocused",
+      "be displayed in viewport": "waitForElementDisplayedInViewport",
+    };
+    return CONDITIONS[condition];
```

Comment:
what about to add negation to this param type and return function that performs certain wait.

Reply:
removed